extensions/cactus-plugin-htlc-coordinator/src/main/typescript/plugin-htlc-coordinator.ts

```diff
@@ -0,0 +1,353 @@
+    counterpartyHTLCRequest: CounterpartyHTLCRequest,
+  ): Promise<InvokeContractV1Response> {
+    const fnTag = `${this.className}#counterpartyHTLC()`;
+    const connector = this.pluginRegistry.plugins.find(
+      (plugin) => plugin.getInstanceId() == counterpartyHTLCRequest.connectorId,
+    ) as PluginLedgerConnectorBesu;
+    const pluginRegistry = new PluginRegistry({});
+    pluginRegistry.add(connector);
+
+    switch (counterpartyHTLCRequest.htlcPackage) {
+      case HtlcPackage.BesuErc20: {
+        const pluginOptions: IPluginHtlcEthBesuErc20Options = {
+          instanceId: uuidv4(),
+          pluginRegistry,
+        };
+
+        const factoryHTLC = new PluginFactoryHtlcEthBesuErc20({
+          pluginImportType: PluginImportType.Local,
+        });
+        const pluginHtlc = await factoryHTLC.create(pluginOptions);
+
+        const res = await pluginHtlc.getSingleStatus(
+          counterpartyHTLCRequest.htlcId,
+          counterpartyHTLCRequest.connectorId,
+          counterpartyHTLCRequest.keychainId,
+          counterpartyHTLCRequest.web3SigningCredential,
+        );
+        return res;
+      }
+      case HtlcPackage.Besu: {
+        const pluginOptions: IPluginHtlcEthBesuOptions = {
+          instanceId: uuidv4(),
+          pluginRegistry,
+        };
+        const factoryHTLC = new PluginFactoryHtlcEthBesu({
+          pluginImportType: PluginImportType.Local,
+        });
+        const pluginHtlc = await factoryHTLC.create(pluginOptions);
+        const res = await pluginHtlc.getSingleStatus(
+          counterpartyHTLCRequest.htlcId,
+          counterpartyHTLCRequest.connectorId,
+          counterpartyHTLCRequest.keychainId,
+          counterpartyHTLCRequest.web3SigningCredential,
+        );
+        return res;
+      }
+      default: {
+        throw new Error(
+          `${fnTag} Unrecognized HTLC Package: ` +
+            `${counterpartyHTLCRequest.htlcPackage} Supported ones are: ` +
+            `${Object.values(HtlcPackage).join(";")}`,
+        );
+      }
+    }
+  }
+  public async withdrawCounterparty(
+    withdrawCounterpartyRequest: WithdrawCounterpartyRequest,
+  ): Promise<InvokeContractV1Response> {
+    const fnTag = `${this.className}#withdrawCounterparty()`;
+    const connector = this.pluginRegistry.plugins.find(
+      (plugin) =>
+        plugin.getInstanceId() == withdrawCounterpartyRequest.connectorId,
+    ) as PluginLedgerConnectorBesu;
+    const pluginRegistry = new PluginRegistry({});
+    pluginRegistry.add(connector);
+
+    switch (withdrawCounterpartyRequest.htlcPackage) {
+      case HtlcPackage.BesuErc20: {
+        const pluginOptions: IPluginHtlcEthBesuErc20Options = {
+          instanceId: uuidv4(),
+          pluginRegistry,
+        };
+
+        const factoryHTLC = new PluginFactoryHtlcEthBesuErc20({
+          pluginImportType: PluginImportType.Local,
+        });
+        const pluginHtlc = await factoryHTLC.create(pluginOptions);
```

Comment:
@jagpreetsinghsasan Please refactor so that the existing plugin registry is used to look up plugin instances instead of creating one from scratch (tests will have to be updated to pass in a plugin registry via the constructor as well - normally the API server takes care of this)